book/src/redundancy.md

```diff
@@ -0,0 +1,106 @@
+`http://localhost:5052` to perform duties. If that node is not contactable, not
+synced or unable to serve the request then the validator client may fail to
+perform some duty (e.g., produce a block or attest).
+
+However, in the second example, any failure on `http://localhost:5052` will be
+followed by a second attempt using `http://192.168.1.1:5052`. This
+achieves *redundancy*, allowing the validator client to continue to perform its
+duties as long as *at least one* of the beacon nodes is available.
+
+There are a few interesting properties about the list of `--beacon-nodes`:
+
+- *Ordering matters*: the validator client prefers a beacon node that is
+	earlier in the list.
+- *Synced is preferred*: the validator client prefers a synced beacon node over
+	one that is still syncing.
+- *Failure is sticky*: if a beacon node fails, it will be flagged as offline
+	the rest of the slot (12 seconds). This helps prevent the impact of time-outs and
+	other lengthy errors.
+
+### Configuring a redundant Beacon Node
+
+In our previous example we listed `http://192.168.1.1:5052` as a redundant
+node. Apart from having sufficient resources, the backup node should have the
+following flags:
+
+- `--staking`: starts the HTTP API server and ensures the Eth1 chain is synced.
+- `--http-address 0.0.0.0`: this allows *any* external IP address to access the
+	HTTP server (a firewall should be configured to deny unauthorized access to port
+	`5052`). This is only required if your backup node is on a different host.
+- `--subscribe-all-subnets`: ensures that the beacon node subscribes to *all*
+	subnets, not just on-demand requests from validators.
+- `--process-all-attestations`: ensures that the beacon node performs
+	aggregation on all seen attestations.
+
+Subsequently, one could use the following command to provide a backup beacon
+node:
+
+```bash
+lighthouse bn \
+  --staking \
+  --http-address 0.0.0.0 \
+  --subscribe-all-subnets \
+  --process-all-attestations
+```
+
+### Resource usage of redundant Beacon Nodes
+
+The `--subscribe-all-subnets` and `--process-all-attestations` flags typically
+cause a significant increase in resource consumption. A doubling in CPU
+utilization and RAM consumption is expected.
+
+The increase in resource consumption is due to the fact that the beacon node is
+now processing, validating, aggregating and forwarding *all* attestations,
+whereas previously it was likely only doing a fraction of this work. Without
+these flags, subscription to attestation subnets and aggregation of
+attestations is only performed for validators which [explicitly request
+subscriptions](subscribe-api).
+
+There are 64 subnets and each validator will result in a subscription to *at
+least* one subnet. So, using the two aforementioned flags will result in
+resource consumption akin to running 64+ validators.
+
+## Redundant Eth1 nodes
+
+Compared to redundancy in beacon nodes (see above), using redundant Eth1 nodes
```

Comment:
Something that I think is worth pointing out:  if you use the `--eth1-endpoints` flag, there is no default fallback to `http://localhost:8545`, so you have to be explicit with localhost whenever you use this flag.   I don't know if this is the same for `--beacon-nodes`

Reply:
Good call, I added some notes :)